packages/visual-editor/src/components/RootRenderer/sendCanvasGeometryUpdatedMessage.ts

```diff
@@ -0,0 +1,80 @@
+import { sendMessage, getElementCoordinates } from '@contentful/experiences-core';
+import { OUTGOING_EVENTS } from '@contentful/experiences-core/constants';
+import {
+  CanvasGeometryUpdateSourceEvent,
+  ExperienceTree,
+  ExperienceTreeNode,
+} from '@contentful/experiences-core/types';
+
+/**
+ * This function gets the element co-ordinates of a specified component in the DOM and its parent
+ * and sends the DOM Rect to the client app
+ */
+export const sendCanvasGeometryUpdatedMessage = async (
+  tree: ExperienceTree,
+  sourceEvent: CanvasGeometryUpdateSourceEvent,
+) => {
+  const nodeRecords: Record<
+    string,
+    { coordinates: Pick<DOMRect, 'x' | 'y' | 'width' | 'height'> }
+  > = {};
+  const scrollX = window.scrollX;
+  const scrollY = window.scrollY;
+
+  await waitForAllImagesToBeLoaded();
+
+  const getNodeCoordinates = (node: ExperienceTreeNode) => {
```

Comment:
nit: maybe for readability it can be declared outside `sendCanvasGeometryUpdatedMessage` function